ouroboros-network/test/Test/Mux.hs

```diff
@@ -148,6 +156,51 @@ newtype DummyPayload = DummyPayload {
 instance Show DummyPayload where
     show d = printf "DummyPayload %d\n" (BL.length $ unDummyPayload d)
 
+-- |
+-- Generate a byte string of a given size.
+--
+genByteString :: Int64 -> Gen BL.ByteString
+genByteString size = MkGen $ \(QCGen g0) _ ->
+  if size <= 0
+    then BL.empty
+    else BL.unfoldr gen (size, g0)
+  where
+    gen :: (Int64, SM.SMGen) -> Maybe (Word8, (Int64, SM.SMGen))
+    gen (!i, !g)
+        | i <= 0    = Nothing
+        | otherwise = Just (fromIntegral w64, (i - 1, g'))
+      where
+        ~(w64, g') = SM.nextWord64 g
```

Comment:
`~` Do we need this? `where`-clause pattern matches are already lazy. Indeed wouldn't `!` be more appropriate?

Reply:
Indeed.